atlasdb-client/src/test/java/com/palantir/atlasdb/util/MetricNameUtilsTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.util;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+
+public class MetricNameUtilsTest {
+
+    @Test
+    public void shouldReturnNameWithSingleTag() throws Exception {
+        String metricName = MetricNameUtils.getMetricName("metricName", ImmutableMap.of("tag1", "tagVal1"));
+        assertThat(metricName).isEqualTo("metricName;tag1=tagVal1");
+    }
+
+    @Test
+    public void shouldReturnNameWithMultipleTags() throws Exception {
+        String metricName = MetricNameUtils.getMetricName("metricName", ImmutableMap.of("tag1", "tagVal1", "tag2", "tagVal2"));
+        assertThat(metricName).isEqualTo("metricName;tag1=tagVal1;tag2=tagVal2");
```

Comment:
Pretty nitpicky, but I'd prefer if we did `contains` on both equality pairs (or it being equal to either this or the line with the tag order flipped). This is safe, but only because we're relying on the `ImmutableMap` preserving iteration order - if you pass a `HashMap`, say, there's no guarantee this test will pass because you could get `metricName;tag2=tagVal2;tag1=tagVal1`.